src/client-agent/config.c

```diff
@@ -126,17 +126,31 @@ cJSON *getLabelsConfig(void) {
         return NULL;
     }
 
-    unsigned int i;
     cJSON *root = cJSON_CreateObject();
-    cJSON *labels = cJSON_CreateObject();
+    cJSON *labels = cJSON_CreateArray();
+    cJSON *label_block = cJSON_CreateObject();
+    cJSON *label_all = cJSON_CreateArray();
 
     if (agt->labels) {
-        for (i=0;agt->labels[i].key;i++) {
-            cJSON_AddStringToObject(labels,agt->labels[i].key,agt->labels[i].value);
+        unsigned int i;
+        for (i=0; agt->labels[i].key; i++) {
+            cJSON *label = cJSON_CreateObject();
+            cJSON_AddStringToObject(label, "value", agt->labels[i].value);
+            cJSON_AddStringToObject(label, "key", agt->labels[i].key);
+
+            if (agt->labels[i].flags.hidden) {
```

Comment:
Change to `cJSON_AddStringToObject(label, "hidden", agt->labels[i].flags.hidden ? "yes" : "no");`